src/levelset/levelSetSegmentationObject.cpp

```diff
@@ -1768,6 +1768,119 @@ int LevelSetSegmentationBaseObject::evalCellPart(long id) const
     return part;
 }
 
+/*!
+ * Check if the specified cell interface intersects the zero-levelset iso-surface.
+ * If an intersection is detected, the centroid and area of the intersected interface is computed.
+ *
+ * @param[in] id cell id
+ * @param[in] intrLocalId is the local index of the interface for the given cell
+ * @param[in] tolerance is the tolerance used for distance comparisons
+ * @param[in] signedLevelSet controls if signed levelset function will be used
+ * @param[in] tolerance is the tolerance used for distance comparisons
+ * @param[out] centroid is the centroid the intersected face computed as the arithmetic avretage of the face vertiecs
+ * @param[out] area is the area of the intersected face
+ * @param[out] intersectionCentroid is the centroid of the intersection computed as the arithmetic avretage of the intersection points
+ * intersection points. If the interface is not intersected, the maximum numeric limit is returned
+ * @return indicator regarding intersection
+ */
+LevelSetIntersectionStatus LevelSetSegmentationBaseObject::_evalIntersectedInterfaceData(long id, int intrLocalId, bool signedLevelSet, double tolerance, std::array<double, 3> *centroid, double *area, std::array<double, 3> *intersectionCentroid) const
```

Comment:
Could this function return only the intersection points and leave the task of evaluating other information to the caller (it can be an overload of _intersectInterfaceSurface)? I mean, centroid and intersectionCentroid seems very specific to the application you are developing (i.e., cut cell method in fluid-dynamic solver). The levelset module doesn't seem the correct place for evaluating this information. The levelset should only provide the intersection between a cell interface and the levelset-zero.